test.js

```diff
@@ -152,6 +146,43 @@ describe('url-parse', function () {
     assume(parsed.pathname).equals('/b/c');
   });
 
+  describe('protocol', function() {
+    it('extracts the right protocol from a url', function() {
+      var testData = [
+        { url: 'http://example.com', protocol: 'http:' },
+        { url: 'mailto:test@example.com', protocol: 'mailto:' },
+        { url: 'data:text/html,%3Ch1%3EHello%2C%20World!%3C%2Fh1%3E', protocol: 'data:' },
+        { url: 'sip:alice@atlanta.com', protocol: 'sip:' }
+      ];
+
+      var data;
+      for (var i = 0, len = testData.length; i < len; ++i) {
```

Comment:
Maybe use `testData.forEach`, it's not very important but I find it easier to grok.